command/format/diff.go

```diff
@@ -1266,3 +1454,11 @@ func DiffActionSymbol(action plans.Action) string {
 		return "  ?"
 	}
 }
+
+// Extremely coarse heuristic for determining whether or not a given attribute
+// name is important for identifying a resource. In the future, this may be
+// replaced by a flag in the schema, but for now this is likely to be good
+// enough.
+func identifyingAttribute(name string, attrSchema *configschema.Attribute) bool {
+	return name == "id" || name == "tags" || name == "name"
```

Comment:
I'm curious why you're leaving in tags here as an unsuppressed attribute - in my experience, configurations using tags use a _lot_ of tags, and those tags tend to be duplicated all over the place. I'm more than happy for this to remain, it's not an objection -  better to hear what users think - but I am curious!

Reply:
This (like most of the good parts of this PR) came from my understanding an idea proposed by @apparentlymart. My very limited experience suggested that `tags` are often used to identify grouped resources, and so it made sense to me to include those in the pass-list of attributes used to identify which resources are changing.

I hadn't considered that `tags` maps/lists might easily be a large part of the diff. Do you think it's worth removing `tags` from this initial implementation, and adding them back if user feedback suggests it? I think it'll be easier to add to this list than remove from it.

Reply:
I don't know ! I only have my own experience to fall back on here. But if you think it's going to be easier to add then remove, I would suggest suppressing tags for the first (dev) release or so and seeing what kind of feedback you get. 

At my last job, we had about 10 required tags (because ENTERPRISE!! 💰 💸  💵 ) on every resource on top of whatever tags the developer added, and so displaying the tags would make the diff not concise at all.

Reply:
And that sounds like _the opposite_ of what we're going for. Removed in 69840e99b.

Reply:
This is definitely a tradeoff with no easy answer: in many cases in AWS the `Name` tag is the only useful (by which I mean: human-meaningful) identifier for an object, and so I'd considered it important to show at least _that_ tag, but because we're currently using an approximation to allow us to ship this ahead of possible future provider protocol changes we can't really put an AWS-specific assumption like special-casing a tag called `Name` in here.

On the other hand, as you note @mildwonkey there are likely to be a number of other values in there that are _not_ useful identifiers for this specific object, and so the resulting output will be noisier than it would ideally be.

My sense here is that showing the `tags` in full while hiding everything else is _still_ considerably more concise than the current behavior for commonly-used AWS provider resource types, and so I feel it's an acceptable compromise until we're ready to design and implement a provider protocol extension to allow the providers themselves to indicate what they consider to be "identifying attributes".

The alternative of hiding `tags` would make it more likely that a particular object in the diff would have no visible human-readable identifiers at all, which was the problem that the non-concise diff change was originally attempting to solve. For that reason, I'd rather err on the side of being not quite as concise in order to decrease the likelihood of producing a diff that can't be understood without referring to an external system (because in that case, it's very tempting to just trust that the objects shown in the diff are the correct objects, without actually referring to the external system to confirm that).

Reply:
I've dropped the prior commit, so we're back to always rendering `tags`.